engineio/src/packet.rs

```diff
@@ -1,13 +1,36 @@
 use base64::{engine::general_purpose, Engine as _};
 use bytes::{BufMut, Bytes, BytesMut};
+use rmp::{decode, encode};
 use serde::{Deserialize, Serialize};
 use std::char;
-use std::convert::TryFrom;
-use std::convert::TryInto;
 use std::fmt::{Display, Formatter, Result as FmtResult, Write};
 use std::ops::Index;
 
 use crate::error::{Error, Result};
+
+pub mod normal;
+pub mod message_pack;
+
+// /// Serializer of Engine.IO packet
+// #[derive(Copy, Clone, Debug, Eq, PartialEq)]
+// pub enum PacketSerializer {
+//     /// Normal serializer
+//     Normal,
+//     /// MessagePack serializer
+//     MessagePack,
+// }
+
+// impl Default for PacketSerializer {
+//     fn default() -> Self {
+//         PacketSerializer::Normal
+//     }
+// }
+
+pub trait PacketSerializer{
```

Comment:
The trait is much better (in case the commented code presents an alternative), although I am not sure if I am 100% happy with the definition. I am going to think about it :)